src/waveform/renderers/waveformbeat.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <QPainter>
+
+#include "preferences/beatgridmode.h"
+#include "proto/beats.pb.h"
+#include "track/beat.h"
+#include "waveform/renderers/waveformelementrightclickable.h"
+
+class WaveformBeat final : WaveformElementRightClickable {
+  public:
+    WaveformBeat();
+    void draw(QPainter* painter) const;
+    void setPositionPixels(int position) {
+        m_position = position;
+    }
+    void setOrientation(Qt::Orientation orientation) {
+        m_orientation = orientation;
+    }
+    void setLength(int length) {
+        m_length = length;
+    }
+    void setBeatGridMode(BeatGridMode mode) {
+        m_beatGridMode = mode;
+    }
+    void setBeat(mixxx::Beat beat) {
+        m_beat = beat;
+    }
+    mixxx::Beat getBeat() const {
+        return m_beat;
+    }
+    void setVisible(bool bVisible) {
+        m_bVisible = bVisible;
+    }
+    void setAlpha(int alpha) {
+        m_iAlpha = alpha;
+    }
+    bool contains(QPoint point, Qt::Orientation orientation) const override;
+
+  private:
+    mixxx::Beat m_beat;
+    Qt::Orientation m_orientation;
+    int m_position;
+    int m_length;
+    int m_iAlpha;
+    BeatGridMode m_beatGridMode;
+    bool m_bVisible;
+};
+
+bool operator<(const WaveformBeat& beat1, const WaveformBeat& beat2);
```

Comment:
Is this function really needed? Defining only a single, selected comparison operator is bad style.

Reply:
This serves as the default comparator in std algorithms that require ordering e.g. `std::lower_bound`

Reply:
Ok, I suppose it makes sense to define the other comparison operators too then?

Reply:
The STL uses `std::less` as the default functor for strict ordering. The default implementation of `std::less` then invokes `operator<`.

It may be more appropriate to instead specialize `std::less` instead of implementing only one of the comparison operators.

Reply:
Not sure though how to correctly specialize `std::less` :grimacing: Even the Core Guidelines don't mention how to handle this situation correctly.

Or just leave a comment that this only needed for the default ordering comparator std::less.